extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -770,125 +350,222 @@ export async function newDeployment(
       "Scanning::newDeployment",
       viewId,
-      async () => await getConfigurationInspectionQuickPicks(entryPointPath),
+      async () =>
+        await getConfigurationInspectionQuickPicks(entryPointFilePath),
     );
 
     // skip if we only have one choice.
     if (inspectionQuickPicks.length > 1) {
-      const step = getStepInfo("inputEntryPointContentTypeSelection", state);
-      if (!step) {
-        window.showErrorMessage(
-          "Internal Error: newDeployment::inputEntryPointContentTypeSelection step info not found.",
-        );
-        return;
+      if (newDeploymentData.entrypoint.inspectionResult) {
+        inspectionQuickPicks.forEach((pick) => {
+          if (
+            pick.inspectionResult &&
+            newDeploymentData.entrypoint.inspectionResult
+          ) {
+            pick.picked = areInspectionResultsSimilarEnough(
+              pick.inspectionResult,
+              newDeploymentData.entrypoint.inspectionResult,
+            );
+          }
+        });
       }
 
       const pick = await input.showQuickPick({
         title: state.title,
-        step: step.step,
-        totalSteps: step.totalSteps,
-        placeholder: `Select the content type for your entrypoint file (${entryPointPath}).`,
+        step: 0,
+        totalSteps: 0,
+        placeholder: `Select the content type for your entrypoint file (${newDeploymentData.entrypoint.filePath}).`,
         items: inspectionQuickPicks,
         buttons: [],
         shouldResume: () => Promise.resolve(false),
         ignoreFocusOut: true,
       });
 
-      state.data.entryPoint = pick;
+      if (!pick || !isQuickPickItemWithInspectionResult(pick)) {
+        return;
+      }
+
+      newDeploymentData.entrypoint.inspectionResult = pick.inspectionResult;
+      return (input: MultiStepInput) => inputContentType(input, state);
+    } else {
+      newDeploymentData.entrypoint.inspectionResult =
+        inspectionQuickPicks[0].inspectionResult;
+      // We're skipping this step, so we must silently just jump to the next step
+      return inputContentType(input, state);
+    }
+  }
+
+  // ***************************************************************
+  // Step: Input the Content Type, if inspection returns UNKNOWN content type
+  // ***************************************************************
+  async function inputContentType(
+    input: MultiStepInput,
+    state: MultiStepState,
+  ) {
+    if (!newDeploymentData.entrypoint.inspectionResult) {
+      return;
+    }
+    if (
+      newDeploymentData.entrypoint.inspectionResult.configuration.type ===
+      ContentType.UNKNOWN
+    ) {
+      // have to prompt user for a content type, since we were unable to determine it.
+      const quickPicks: QuickPickItemWithIndex[] = [];
+      allValidContentTypes.forEach((contentType, index) => {
+        quickPicks.push({
+          label: contentType,
+          detail: contentTypeStrings[contentType],
```

Comment:
I found this style to be a bit large given the strings we use. I believe a swap from `detail` to `description` would be a bit more dense, and easier to parse:

![CleanShot 2024-09-25 at 15 12 39@2x](https://github.com/user-attachments/assets/1b80d7ce-111e-4d10-99c8-001495cf1b1b)